vaultclient/vaultclient.go

```diff
@@ -1,63 +1,177 @@
 package vaultclient
 
 import (
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/client"
-	"github.com/aws/aws-sdk-go/aws/client/metadata"
-	"github.com/aws/aws-sdk-go/aws/request"
-	"github.com/aws/aws-sdk-go/aws/signer/v4"
-	"github.com/aws/aws-sdk-go/private/protocol/query"
-	"github.com/aws/aws-sdk-go/private/protocol/restjson"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
 )
 
 // Vault provides the API operation methods for making requests to Vault.
 type Vault struct {
-	*client.Client
+	cfg      aws.Config
+	client   *http.Client
+	endpoint string
 }
 
 // Service information constants
 const (
-	ServiceName = "iam"       // Name of service.
-	EndpointsID = ServiceName // ID to lookup a service endpoint with.
-	ServiceID   = "Vault"     // ServiceID is a unique identifier of a specific service.
+	ServiceName = "iam" // Name of service.
 )
 
-// New creates a new instance of the Vault client with a session.
-func New(p client.ConfigProvider, cfgs ...*aws.Config) *Vault {
-	c := p.ClientConfig(EndpointsID, cfgs...)
-	return newClient(*c.Config, c.Handlers, c.PartitionID, c.Endpoint, c.SigningRegion, c.SigningName)
-}
-
-func newClient(cfg aws.Config, handlers request.Handlers, partitionID, endpoint, signingRegion, signingName string) *Vault {
-	svc := &Vault{
-		Client: client.New(
-			cfg,
-			metadata.ClientInfo{
-				ServiceName:   ServiceName,
-				ServiceID:     ServiceID,
-				SigningName:   signingName,
-				SigningRegion: signingRegion,
-				PartitionID:   partitionID,
-				Endpoint:      endpoint,
-				APIVersion:    "2010-05-08",
-			},
-			handlers,
-		),
+func New(accessKey, secretKey, sessionToken, endpoint, region string) *Vault {
+	return &Vault{
+		cfg: aws.Config{
```

Comment:
since we use aws.Config,
* might as well pass it as argument to New (as before)
* we may use the `BaseEndpoint` and `HTTPClient` field directly, instead of adding our own?